composer.json

```diff
@@ -29,7 +29,7 @@
         "doctrine/doctrine-bundle": "^2.3",
         "doctrine/orm": "^2.5",
         "doctrine/persistence": "^2.0",
-        "sonata-project/admin-bundle": "4.0.0-alpha-1",
+        "sonata-project/admin-bundle": "^4.0",
```

Comment:
Also we should keep in mind that the 4.0-alpha-2 won't work with every futur alpha.
So maybe we should add a conflict for the futur alpha ?

Reply:
I dont understand, please take a look at the build, there is already one checking against dev master and it is working just fine

Reply:
This is my issue:
Let's say we're merging https://github.com/sonata-project/SonataAdminBundle/pull/7022, it introduce a BC-break and a new feature. How can we add compatibility in the persistence bundle ?
- If we only use `^4.0`, we cannot use the last commit of SonataAdmin.
- If we release an alpha-2 from SonataAdmin, during the time we add compatibility, people could load the alpha2 from SonataAdmin with the alpha1 from this bundle.

So how do we do ?

Reply:
I'm not sure this PR is fixing the right thing.
We shouldn't allow more alpha than the alpha 1 since we cannot be sure it will work for others alpha.

If I understand correctly we want to fix the build 
```
Test / PHP 8.0 + highest + sonata-project/admin-bundle:"dev-master" (pull_request)
```
from another bundle.

Maybe the only change to do is something like `4.0.0-alpha-1 || dev-master` ?
Or maybe it's just an issue from the other bundle, which should run 
```
Test / PHP 8.0 + highest + sonata-project/admin-bundle:"4.0.0-alpha-1" (pull_request)
```
instead of dev-master

Reply:
> This is my issue:
> 
> Let's say we're merging https://github.com/sonata-project/SonataAdminBundle/pull/7022, it introduce a BC-break and a new feature. How can we add compatibility in the persistence bundle ?
> 
> - If we only use `^4.0`, we cannot use the last commit of SonataAdmin.
> 
> - If we release an alpha-2 from SonataAdmin, during the time we add compatibility, people could load the alpha2 from SonataAdmin with the alpha1 from this bundle.
> 
> 
> 
> So how do we do ?

If you merge a pr that breaks bc, this master branch will start to fail on the dev-master build. Then when you want to fix it, you should add a conflict with alpha1 and only let install dev-master (through minimum stability dev)

Reply:
Ok, so we're gonna update often this composer json.

Reply:
I think so